### DISCUSSION TITLE
Shell Depth part

### DISCUSSION THREAD

**Discussion by Lockszmith-GH** [JanDeDobbeleer/oh-my-posh#2970] [Troubleshoot] [answered]
# Works in Linux
I came up with the following segment to indicate shell depth:
```
        {
          "type": "command",
          "foreground": "#d3d7cf",
          "background": "black",
          "powerline_symbol": "",
          "style": "powerline",
          "properties": {
            "shell": "bash",
            "command": "pstree -s $$ | head -1 | sed -r 's/-\\+-/---/g; s/-[^-]+---oh-my-posh---.*/-/; s/---/\\n/g; ' | grep -Ec \"^($( cat /etc/shells | tail +2 | sed 's/.*\\///' | sort -u | grep -v tmux | paste -sd '|' ))$\""
          },
          "template": "{{ if not ( eq .Output \"1\" ) }}{{ .Output }}{{ end }}"
        }
```

# Question
This will work well in Linux, any idea how to make this work in Windows?
Is there a builtin section that does this?

# Not part of the question
### How does this work (for those finding the code in the question of interest)
I'm running a `bash` shell command line that counts the number of shell process names in the pstree of the current process.
* `pstree -s $$` - this retrieves the root of the current process ID (`$$` in `bash`)
* `head -1` - only grab the first line (sed usually forks, and shows up as a second line, I don't need that line)
* Using `sed`:
  * `s/-\+-/---/g;` - make sure all connectors are marked as three-dashes ('`---`')
  * `s/-[^-]+---oh-my-posh---.*/-/; ` - Clear everything past oh-my-posh
  * `s/---/\n/g;` - finally, split the list into lines.
* `grep -Ec ...` - Use extended-regexp (EREs) syntax, and count the number of matched lines. `...` here is:
  * '"^$(...)$"' - match the entire line (whole words only), where `...` here is the following that grabs all valid shell names.
    The result of this command looks something like: `bash|dash|pwsh|rbash|sh|zsh`
    * `cat /etc/shells | tail +2` - list all valid shell names, starting with the 2nd line (1st line is just a title)
    * `sed 's/.*\\///'` - keep only the last part after '/'
    * `sort -u` - sort, and keep only unique lines
    * `grep -v tmux` - remove any line with tmux - I don't want to count that as a shell level
    * `paste -sd '|'` - combine all lines into a single string delimited by `|`

**Comment by Lockszmith-GH** [comment, marked as answer]
I ended up simplifying this process.
My shell initializes an Environment variable, and the prompt just shows the var if it's more than 1.

If anyone has another solution, I'm interested.

**Comment by SiJiL82** [comment]
I've been after something similar, been using `$SHLVL`
It always reports higher than running `echo $SHLVL` in the same terminal though - sometimes 1 higher but in VSCode terminal, 2 higher.
